pkg/controller.go

```diff
@@ -37,29 +40,42 @@ type Controller struct {
 	logr.Logger
 }
 
-func (c *Controller) Update(csi *csibaremetalv1.Deployment) error {
+func (c *Controller) Update(csi *csibaremetalv1.Deployment, scheme *runtime.Scheme) error {
+	// create deployment
+	expected := createControllerDeployment(csi)
+	if err := controllerutil.SetControllerReference(csi, expected, scheme); err != nil {
+		return err
+	}
+
 	namespace := GetNamespace(csi)
 	dsClient := c.AppsV1().Deployments(namespace)
 
-	isDeployed, err := isDeploymentDeployed(dsClient, controllerName)
+	found, err := dsClient.Get(controllerName, metav1.GetOptions{})
 	if err != nil {
-		c.Logger.Error(err, "Failed to get daemon set")
+		if errors.IsNotFound(err) {
+			if _, err := dsClient.Create(expected); err != nil {
+				c.Logger.Error(err, "Failed to create deployment")
+				return err
+			}
+
+			c.Logger.Info("Deployment created successfully")
+			return nil
+		}
+
+		c.Logger.Error(err, "Failed to get deployment")
 		return err
 	}
 
-	if isDeployed {
-		c.Logger.Info("Deployment already deployed")
-		return nil
-	}
+	if deploymentChanged(expected, found) {
```

Comment:
Seems like you've solved my issue with upgrade:)

Reply:
Yes, I think, it should work. It is no difference if expected or found instance has been changed

Reply:
@AnastasiaShemyakinskaya `deploymentChanged` doesn't check version